Server/mods/deathmatch/logic/CElementDeleter.cpp

```diff
@@ -18,11 +18,17 @@ void CElementDeleter::Delete(class CElement* pElement, bool bUnlink, bool bUpdat
     if (pElement)
     {
         if (!IsBeingDeleted(pElement))
-        {
+        {            
             // Before we do anything, fire the on-destroy event
             CLuaArguments Arguments;
             pElement->CallEvent("onElementDestroy", Arguments);
 
+            // Check if the element is a team and remove players
+            if (IS_TEAM(pElement))
+            {
+               ((CTeam*)pElement)->RemoveAllPlayers();
```

Comment:
Maybe a `static_cast` would look better(and be more proper) here:
`static_cast<CTeam*>(pElement)->RemoveAllPlayers())`.